### DISCUSSION TITLE
RPi5 M.2 NVMe SSD with BirdNET-Pi

### DISCUSSION THREAD

**Discussion by dalogue1** [Nachtzuster/BirdNET-Pi#98] [General]
Is anyone using an RPi5 4GB and an M.2 NVMe SSD with BirdNET-Pi?

I am setting it up this weekend and wanted to know if there are any gotchas.

Current expected setup is with the following:

- RPi5 4GB
- RPi M.2 Hat
- 256GB WD M.2 NVMe SSD
- AOM-5024 Homemade microphone
- SABRENT USB Audio Adapter
- 20W USBC Power adapter
- 256GB microSD for setup

**Comment by noiselevelzero** [comment]
I didn't see anyone answer your question, but I had a spare PI5 with a 256GB SSD here, so built it with the Nachtzuster BirdnetPi fork to replace my Pi4 and currently have it testing with a lavaliere microphone.  It's not as good as the Mic I have on my current Birdnetpi, but works pretty well regardless.

Verdict?  Other than the fact that I had to take the SSD and put it into an M2 USB reader to use the Raspberry Pi Imager to install Bookworm 64-bit lite, it was relatively turnkey.  Even though I wasn't sure I had to, I did run the disk expansion using SUDO RASPI-CONFIG, and it dutifully rebooted.  All the menus appear to work, but have not yet started it feeding to Birdweather.  I see all of the card available in System Info, and am logging plenty of birds.  It's quite peppy, but other than that and the much expanded storage, I don't see a lot of difference in how it works.  Time will tell.  Tonight, after all the birds go to bed around here, I'll pull my current working pi, and move the database files across from the SD card to this new unit.   I appreciate your question, which got me thinking about how hard this system must be pounding my (industrial) microsd card, and how much more storage I'd gain with the SSD.

Short answer, it works, and I don't anticipate you will have any trouble getting it working.

Cheers.

**Reply by dalogue1** [reply]
Well while waiting on getting the M.2 hat, I set up the system using a 1TB SSD using USB3.  I then played a recording from the Merlin app that recognized 1) an **American Robin** 2) a **White winged Dove** 3) a Northern Cardinal and 4) a **Scaly Breasted Munia**. (all which I could physically see)

I can see from the following log that it is seeing three of the the birds but none are listed in the "Overview" or "Today's Detection"

What am I doing wrong?  Do I need to change a setting to start logging recognized birds?
 
1:42:25---[birdnet_analysis][INFO] Analyzing BirdSongs/StreamData/2024-06-15-birdnet-01:42:10.wav
01:42:25---[server][INFO] READING AUDIO DATA...
01:42:25---[server][INFO] READING DONE! READ 5 CHUNKS.
01:42:25---[server][INFO] ANALYZING AUDIO...
01:42:26---[server][INFO] DONE! Time 0.53 SECONDS
01:42:26---[server][INFO] 0.0;3.0-('Human_Human', 0.0)
01:42:26---[server][INFO] 3.0;6.0-('Basileuterus rufifrons_Rufous-capped Warbler', 0.05121438)
01:42:26---[server][INFO] 6.0;9.0-('Human_Human', 0.0)
01:42:26---[server][INFO] 9.0;12.0-('Human_Human', 0.0)
01:42:26---[server][INFO] 12.0;15.0-('Human_Human', 0.0)
01:42:40---[birdnet_analysis][INFO] Analyzing BirdSongs/StreamData/2024-06-15-birdnet-01:42:25.wav
01:42:40---[server][INFO] READING AUDIO DATA...
01:42:40---[server][INFO] READING DONE! READ 5 CHUNKS.
01:42:40---[server][INFO] ANALYZING AUDIO...
01:42:41---[server][INFO] DONE! Time 0.55 SECONDS
01:42:41---[server][INFO] 0.0;3.0-('Human_Human', 0.0)
01:42:41---[server][INFO] 3.0;6.0-('Turdus plumbeus_Red-legged Thrush', 0.41894102)
01:42:41---[server][INFO] 6.0;9.0-('Turdus migratorius_**American Robin**', 0.25159693)
01:42:41---[server][INFO] 9.0;12.0-('Turdus migratorius_American Robin', 0.2077426)
01:42:41---[server][INFO] 12.0;15.0-('Lonchura punctulata_**Scaly-breasted Munia**', 0.5922488)
01:42:55---[birdnet_analysis][INFO] Analyzing BirdSongs/StreamData/2024-06-15-birdnet-01:42:40.wav
01:42:55---[server][INFO] READING AUDIO DATA...
01:42:55---[server][INFO] READING DONE! READ 5 CHUNKS.
01:42:55---[server][INFO] ANALYZING AUDIO...
01:42:56---[server][INFO] DONE! Time 0.54 SECONDS
01:42:56---[server][INFO] 0.0;3.0-('Turdus migratorius_American Robin', 0.11889857)
01:42:56---[server][INFO] 3.0;6.0-('Human_Human', 0.0)
01:42:56---[server][INFO] 6.0;9.0-('Human_Human', 0.0)
01:42:56---[server][INFO] 9.0;12.0-('Human_Human', 0.0)
01:42:56---[server][INFO] 12.0;15.0-('Human_Human', 0.0)
01:43:10---[birdnet_analysis][INFO] Analyzing BirdSongs/StreamData/2024-06-15-birdnet-01:42:55.wav
01:43:10---[server][INFO] READING AUDIO DATA...
01:43:10---[server][INFO] READING DONE! READ 5 CHUNKS.
01:43:10---[server][INFO] ANALYZING AUDIO...
01:43:11---[server][INFO] DONE! Time 0.55 SECONDS
01:43:11---[server][INFO] 0.0;3.0-('Human_Human', 0.0)
01:43:11---[server][INFO] 3.0;6.0-('Coccyzus americanus_Yellow-billed Cuckoo', 0.119108826)
01:43:11---[server][INFO] 6.0;9.0-('Turdus migratorius_American Robin', 0.09483423)
01:43:11---[server][INFO] 9.0;12.0-('Turdus migratorius_American Robin', 0.32546818)
01:43:11---[server][INFO] 12.0;15.0-('Laterallus exilis_Gray-breasted Crake', 0.10740823)
01:43:25---[birdnet_analysis][INFO] Analyzing BirdSongs/StreamData/2024-06-15-birdnet-01:43:10.wav
01:43:25---[server][INFO] READING AUDIO DATA...
01:43:25---[server][INFO] READING DONE! READ 5 CHUNKS.
01:43:25---[server][INFO] ANALYZING AUDIO...
01:43:26---[server][INFO] DONE! Time 0.55 SECONDS
01:43:26---[server][INFO] 0.0;3.0-('Megascops trichopsis_Whiskered Screech-Owl', 0.1447843)
01:43:26---[server][INFO] 3.0;6.0-('Human_Human', 0.0)
01:43:26---[server][INFO] 6.0;9.0-('Human_Human', 0.0)
01:43:26---[server][INFO] 9.0;12.0-('Human_Human', 0.0)
01:43:26---[server][INFO] 12.0;15.0-('Human_Human', 0.0)
01:43:40---[birdnet_analysis][INFO] Analyzing BirdSongs/StreamData/2024-06-15-birdnet-01:43:25.wav
01:43:40---[server][INFO] READING AUDIO DATA...
01:43:40---[server][INFO] READING DONE! READ 5 CHUNKS.
01:43:40---[server][INFO] ANALYZING AUDIO...
01:43:41---[server][INFO] DONE! Time 0.54 SECONDS
01:43:41---[server][INFO] 0.0;3.0-('Human_Human', 0.0)
01:43:41---[server][INFO] 3.0;6.0-('Human_Human', 0.0)
01:43:41---[server][INFO] 6.0;9.0-('Human_Human', 0.0)
01:43:41---[server][INFO] 9.0;12.0-('Human_Human', 0.0)
01:43:41---[server][INFO] 12.0;15.0-('Cyanocitta cristata_Blue Jay', 0.12556545)
01:43:55---[birdnet_analysis][INFO] Analyzing BirdSongs/StreamData/2024-06-15-birdnet-01:43:40.wav
01:43:55---[server][INFO] READING AUDIO DATA...
01:43:55---[server][INFO] READING DONE! READ 5 CHUNKS.
01:43:55---[server][INFO] ANALYZING AUDIO...
01:43:56---[server][INFO] DONE! Time 0.54 SECONDS
01:43:56---[server][INFO] 0.0;3.0-('Human_Human', 0.0)
01:43:56---[server][INFO] 3.0;6.0-('Human_Human', 0.0)
01:43:56---[server][INFO] 6.0;9.0-('Piaya melanogaster_Black-bellied Cuckoo', 0.05487452)
01:43:56---[server][INFO] 9.0;12.0-('Pionites melanocephalus_Black-headed Parrot', 0.14126997)
01:43:56---[server][INFO] 12.0;15.0-('Zenaida asiatica_**White-winged Dove'**, 0.20232545)
01:44:10---[birdnet_analysis][INFO] Analyzing BirdSongs/StreamData/2024-06-15-birdnet-01:43:55.wav
01:44:10---[server][INFO] READING AUDIO DATA...
01:44:10---[server][INFO] READING DONE! READ 5 CHUNKS.
01:44:10---[server][INFO] ANALYZING AUDIO...
01:44:11---[server][INFO] DONE! Time 0.54 SECONDS
01:44:11---[server][INFO] 0.0;3.0-('Human_Human', 0.0)
01:44:11---[server][INFO] 3.0;6.0-('Human_Human', 0.0)
01:44:11---[server][INFO] 6.0;9.0-('Human_Human', 0.0)
01:44:11---[server][INFO] 9.0;12.0-('Human_Human', 0.0)
01:44:11---[server][INFO] 12.0;15.0-('Human_Human', 0.0)
01:44:25---[birdnet_analysis][INFO] Analyzing BirdSongs/StreamData/2024-06-15-birdnet-01:44:10.wav
01:44:25---[server][INFO] READING AUDIO DATA...
01:44:25---[server][INFO] READING DONE! READ 5 CHUNKS.
01:44:25---[server][INFO] ANALYZING AUDIO...
01:44:26---[server][INFO] DONE! Time 0.57 SECONDS
01:44:26---[server][INFO] 0.0;3.0-('Human_Human', 0.0)
01:44:26---[server][INFO] 3.0;6.0-('Human_Human', 0.0)
01:44:26---[server][INFO] 6.0;9.0-('Dryocopus pileatus_Pileated Woodpecker', 0.076771766)
01:44:26---[server][INFO] 9.0;12.0-('Human_Human', 0.0)
01:44:26---[server][INFO] 12.0;15.0-('Human_Human', 0.0)
01:44:40---[birdnet_analysis][INFO] Analyzing BirdSongs/StreamData/2024-06-15-birdnet-01:44:25.wav

**Comment by noiselevelzero** [comment]
Make sure your latitude and longitude are set correctly; the system will not start recognition until it knows where you are.  That said, none of them look like they are meeting the default 70% (.70) threshold for successful recognition.  The highest I see is a .41. Either the audio is not clear, your microphone is not set right, or you need to increase the gain.

I could be missing something however.

**Reply by dalogue1** [reply]
I am not using my AOM-5024 microphone yet.  It is just a cheap USB microphone.  I will check the lat and long setting as you mentioned.

**Comment by noiselevelzero** [comment]
After running my BirdnetPi on a Raspberry Pi 5 and SSD overnight, I've switched back to my Pi4.  I did what I believed was a careful migration of my data to the new Pi's SSD, but the system actually ran slower on the Pi5, and depending on the browser, the refresh rate was terrible.  Also, the "currently analyzing" display on the main view was broken, and only showed an icon.

This may be related to permissions on my migrated birds.db or BirdSongs folder (although I checked), but switching back it is definitely peppier overall for me on a Pi4.

Everything appeared to be working very well on a Pi5 before I migrated any of my backup data from the 4, so for a fresh install this may be just fine, but I want to keep my database which contains a fair amount of spring migration detections.

Good luck.

**Reply by dalogue1** [reply]
Thank you so much for the effort. I am concerned that I have taken the RPi5 path.  Live and learn I guess. 

Anyway I will get it configured with a fresh install and a good microphone and see how it goes.

**Reply by MueJosh** [reply]
That's weird 

I have a pi 5 8gb with nvme base + 500gb (Samsung 970?? Idk) and it works great! Connected via Ethernet and everything is snappy.
Ram drive is off (for now). 

Ugreen soundcard with the standard diy mic (Forgot the capsule name) 
I'm not missing any detections (have a pi 02w and 4b running right next to it)

**Comment by noiselevelzero** [comment]
You are very welcome; it kept me occupied for the evening doing something productive, and gave me a chance to back up all my data and image my card before playing around.  Backups are never a waste of time.  I don't think you will regret the RPI5.  If you start with one you will likely have an excellent experience.  I think I did not do something right when I stopped the services and brought in my old data. 

Let me know how it goes.  Cheers.

**Comment by dalogue1** [comment]
By the way, I am looking forward to it identifying the Scaly Breasted Munia and listing them on BirdNET.  I have at times had as many as 12 of them simultaneously on my bird feeder.  Funny thing is that they are not typically listed as a native bird in Houston.

Funnily enough, in the past the most common bird I had in other homes were brown house sparrows.  I rarely see even one of these in my backyard.  I wonder if the flock of the Scaly Breasted Munias that live in my tree in the backyard keep the sparrows away.  Just a thought.

**Reply by noiselevelzero** [reply]
I hope you do detecte them, but if you do not, you may have to adjust the settings.  Birds that are not typically from your area or not spotted by 2% of the devices in your area (I think that is the default) may be automatically ignored, whether or not they are sitting right next to your microphone.  Others will know better how to make it so you can avoid the Munia being embargoed on the device.  

You will find out that the diversity of bird species even in an urban environment is far greater than you might realize.  I have detected almost 95 different species in and around my suburban lot in just two months using a Birdnet Pi.

**Comment by dalogue1** [comment]
I set the system up temporarily with a USB3 SSD and a Blue Yeti microphone.  It is very quiet right now because it is hot as blazes outside, but it has detected a Northern Cardinal and a Blue Jay both with 89% accuracy.  (The playback sounds verify it since I know what they sound like.)

Very exciting to get it working at all.

**Comment by dalogue1** [comment]
The system just identified the Scaly Breasted Munia at 79% and did not record it.  Now what do I do to get it to recognize it?

I have enough compute power to recognize it but don't accept it? Why?

21:21:01---[server][INFO] READING DONE! READ 5 CHUNKS.
21:21:01---[server][INFO] ANALYZING AUDIO...
21:21:02---[server][INFO] DONE! Time 0.55 SECONDS
21:21:02---[server][INFO] 0.0;3.0-('Lonchura punctulata_Scaly-breasted Munia', 0.6186333)
21:21:02---[server][INFO] 3.0;6.0-('Calidris alpina_Dunlin', 0.25349447)
21:21:02---[server][INFO] 6.0;9.0-('Lonchura punctulata_Scaly-breasted Munia', 0.74261576)
21:21:02---[server][INFO] 9.0;12.0-('Empidonax flaviventris_Yellow-bellied Flycatcher', 0.23393229)
21:21:02---[server][INFO] 12.0;15.0-('Engine_Engine', 0.16423693)
21:21:16---[birdnet_analysis][INFO] Analyzing BirdSongs/StreamData/2024-06-15-birdnet-21:21:01.wav
21:21:16---[server][INFO] READING AUDIO DATA...
21:21:16---[server][INFO] READING DONE! READ 5 CHUNKS.
21:21:16---[server][INFO] ANALYZING AUDIO...

**Comment by noiselevelzero** [comment]
First, go to settings and untick the species range model, and save that.  The species range model is much more conservative, and I've found it makes the system rather deaf.

Then, you are going to have to play with the Species Occurrence Frequency Threshold, which determines what gets recorded for your area based on some previous data on species occurrence.  Others know more about this than I, and if you change it to capture that other bird, it will lead to more false detection's, but you will have to play with that.  That is the extent of my knowledge on the topic.  You may want to start a new question, because it's buried in this thread.

Good luck.

**Comment by dalogue1** [comment]
You are correct.  I needed to start a new thread.

I fixed the problem by lowering the Threshhold to 0.0005

**Reply by dalogue1** [reply]
The OEM WD SN740 256GB does not appear to me compatible.  I get format errors.  I think I will try a 1TB Corsair MP600 instead. 

Apparently there are many NVMe SSDs with issues on the RPi5 M.2 Hat especially Western Digital drives.
